addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -678,16 +678,47 @@ var FileWidget = SearchableMediaWidget.extend({
     /**
      * @private
      */
-    _onURLInputChange: function () {
+    _onURLInputChange: async function() {
         const inputValue = this.$urlInput.val().split('?')[0];
+        const domainExtensions = [
+            'com', 'net', 'org', 'ac', 'ad', 'ae', 'af', 'ag', 'ai', 'al', 'am', 'an',
+            'ao', 'aq', 'ar', 'as', 'at', 'au', 'aw', 'ax', 'az', 'ba', 'bb', 'bd',
+            'be', 'bf', 'bg', 'bh', 'bi', 'bj', 'bl', 'bm', 'bn', 'bo', 'br', 'bq',
+            'bs', 'bt', 'bv', 'bw', 'by', 'bz', 'ca', 'cc', 'cd', 'cf', 'cg', 'ch',
+            'ci', 'ck', 'cl', 'cm', 'cn', 'co', 'cr', 'cs', 'cu', 'cv', 'cw', 'cx',
+            'cy', 'cz', 'dd', 'de', 'dj', 'dk', 'dm', 'do', 'dz', 'ec', 'ee', 'eg',
+            'eh', 'er', 'es', 'et', 'eu', 'fi', 'fj', 'fk', 'fm', 'fo', 'fr', 'ga',
+            'gb', 'gd', 'ge', 'gf', 'gg', 'gh', 'gi', 'gl', 'gm', 'gn', 'gp', 'gq',
+            'gr', 'gs', 'gt', 'gu', 'gw', 'gy', 'hk', 'hm', 'hn', 'hr', 'ht', 'hu',
+            'id', 'ie', 'il', 'im', 'in', 'io', 'iq', 'ir', 'is', 'it', 'je', 'jm',
+            'jo', 'jp', 'ke', 'kg', 'kh', 'ki', 'km', 'kn', 'kp', 'kr', 'kw', 'ky',
+            'kz', 'la', 'lb', 'lc', 'li', 'lk', 'lr', 'ls', 'lt', 'lu', 'lv', 'ly',
+            'ma', 'mc', 'md', 'me', 'mf', 'mg', 'mh', 'mk', 'ml', 'mm', 'mn', 'mo',
+            'mp', 'mq', 'mr', 'ms', 'mt', 'mu', 'mv', 'mw', 'mx', 'my', 'mz', 'na',
+            'nc', 'ne', 'nf', 'ng', 'ni', 'nl', 'no', 'np', 'nr', 'nu', 'nz', 'om',
+            'pa', 'pe', 'pf', 'pg', 'ph', 'pk', 'pl', 'pm', 'pn', 'pr', 'ps', 'pt',
+            'pw', 'py', 'qa', 're', 'ro', 'rs', 'ru', 'rw', 'sa', 'sb', 'sc', 'sd',
+            'se', 'sg', 'sh', 'si', 'sj', 'sk', 'sl', 'sm', 'sn', 'so', 'sr', 'ss',
+            'st', 'su', 'sv', 'sx', 'sy', 'sz', 'tc', 'td', 'tf', 'tg', 'th', 'tj',
+            'tk', 'tl', 'tm', 'tn', 'to', 'tp', 'tr', 'tt', 'tv', 'tw', 'tz', 'ua',
+            'ug', 'uk', 'um', 'us', 'uy', 'uz', 'va', 'vc', 've', 'vg', 'vi', 'vn',
+            'vu', 'wf', 'ws', 'ye', 'yt', 'yu', 'za', 'zm', 'zr', 'zw', 'co\\.uk'];
+
+        const isURL = new RegExp(`^(?:(?:https?|http):\/\/)?(?:\\S+(?::\\S*)?@)?(?:www\\.)?(?:[-\\w.]+)(?::\\d+)?(?:\\/[\\S.]+)?(?:[.](${domainExtensions.join('|')}))?$`).test(inputValue);
         var emptyValue = (inputValue === '');
+        try {
+            if (isURL) {
+                const response = await fetch(inputValue, { method: 'HEAD' });
+                const contentType = response.headers.get('Content-Type');
+                const isImage = this.IMAGE_MIMETYPES.includes(contentType);
+                this._updateAddUrlUi(emptyValue, isURL, isImage);
+            } else {
+                this._updateAddUrlUi(emptyValue, false, false);
+            }
+        } catch (error) {
+            this._updateAddUrlUi(emptyValue, false, false);
+        }
 
-        var isURL = /^.+\..+$/.test(inputValue); // TODO improve
```

Comment:
@dmo-odoo in reference to your recent [comment](https://github.com/odoo/odoo/pull/122118/files#r1317203603) i tried to improve this, previously anything we type on input containing a `.` , say `this.doc.sayHello` the existing regex would have passed it to be a valid url. and the we are needed to perform fetch operation for it which made no sense while fixing.

Reply:
I don't understand the link between that and the original issue but if you really want to do it then you could use our already existing regex for this: https://github.com/odoo/odoo/blob/master/addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js#L54

Don't forget that the entire point of this task is simply to _not_ block a valid image which happens to not have a standard image extension. For that purpose, I think the changes have gotten a bit out of hand.

Reply:
@dmo-odoo i have tried using that regex but it failed for the scenario i mentioned in comment, like typing `this.document.getSelection` that regex conisders it a valid url and fetching it made no sense.

Reply:
Ok. I suggest we refocus this task to its original point, that is simply to not block a valid image which happens to not have a standard image extension, and no other changes.

Reply:
@dmo-odoo made changes.